src/Features/CSharp/Portable/GenerateType/CSharpGenerateTypeService.cs

```diff
@@ -558,6 +567,12 @@ protected override Accessibility GetAccessibility(State state, SemanticModel sem
                 {
                     accessibility = accessibilityConstraint;
                 }
+                else if (accessibilityConstraint == Accessibility.Protected ||
+                         accessibilityConstraint == Accessibility.ProtectedOrInternal)
+                {
+                    // If nested type is declared in public type then we should generate public type instead of internal
+                    accessibility = AllContainingTypesArePublicOrProtected(state, semanticModel, cancellationToken) ? Accessibility.Public : Accessibility.Internal;
```

Comment:
nit: try to wrap lines longer than what github can display.  our general patter for ?: is 

```c#
accessibility = AllContainingTypesArePublicOrProtected(state, semanticModel, cancellationToken)
    ? Accessibility.Public
    : Accessibility.Internal;
```